### PR TITLE
Collocation scheduling

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Adds scheduling to the collocation APIs.
- [x] Reformats data output 

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [AN-325](https://airqoteam.atlassian.net/browse/AN-325)
    - [AN-326](https://airqoteam.atlassian.net/browse/AN-326)
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

[AN-325]: https://airqoteam.atlassian.net/browse/AN-325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AN-326]: https://airqoteam.atlassian.net/browse/AN-326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: k8s/device-monitor/templates/deployment.yaml [hunk 1]
```diff
@@ -59,6 +59,30 @@ spec:
...
+          imagePullPolicy: IfNotPresent
+          envFrom:
+            - configMapRef:
+                name: {{ .Values.app.configmap }} 
+          {{- with .Values.volumeMounts }}
+          volumeMounts:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
```
Comment (NoahNsimbe) [inline k8s/device-monitor/templates/deployment.yaml:85]: @123MwanjeMike

File: src/device-monitoring/Dockerfile [hunk 2]
```diff
@@ -23,3 +23,9 @@ CMD ["gunicorn", "--bind=0.0.0.0:4001", "app:app"]
 FROM base as production
 ENV FLASK_ENV=production
 CMD ["gunicorn", "--bind=0.0.0.0:4001", "app:app"]
+
+FROM base as celery-beat
+CMD ["celery -A app.celery beat"]
+
+FROM base as celery-worker
+CMD ["celery -A app.celery worker --loglevel=info"]
```
Comment (NoahNsimbe) [inline src/device-monitoring/Dockerfile:31]: @123MwanjeMike

File: .github/workflows/deploy-apis-to-production.yml [hunk 3]
```diff
@@ -651,13 +651,29 @@ jobs:
...
-          docker tag ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-device-monitor-api:${{ needs.image-tag.outputs.build_id }} ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-device-monitor-api:latest
-          docker push ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-device-monitor-api:${{ needs.image-tag.outputs.build_id }}
-          docker push ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-device-monitor-api:latest
+      - name: Build and push  API
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          context: src/device-monitoring/
+          target: production
+          tags: ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-device-monitor-api:${{ needs.image-tag.outputs.build_id }},${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/airqo-device-monitor-api:latest
```
Comment (NoahNsimbe) [inline .github/workflows/deploy-apis-to-production.yml:660]: @123MwanjeMike

File: src/device-monitoring/config/constants.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ class Config:
     CACHE_DEFAULT_TIMEOUT = THIRTY_MINUTES
     CACHE_KEY_PREFIX = 'device-monitoring'
     CACHE_REDIS_URL = os.getenv('REDIS_URL_PROD')
+    REDIS_URL = os.getenv('REDIS_URL')
```
Comment (123MwanjeMike) [inline src/device-monitoring/config/constants.py:26]: Do we have this environmental variable registered in our .env files? Is it a new one or a pre-existing one?

**Review by 123MwanjeMike — COMMENTED** [review]
Great work @NoahNsimbe 
I have made a few minor refactoring changes. All is good.
Just a question tho...

File: src/device-monitoring/config/constants.py (hunk 4 above)
Comment (123MwanjeMike) [reply src/device-monitoring/config/constants.py:26]: I've confirmed that we have `REDIS_URL` and also found that we do not have `REDIS_URL_PROD` in our configs. Could we stick to using `REDIS_URL` given that we may need to run the app in other environments other than prod?

**Review by 123MwanjeMike — APPROVED** [review]

**Pull request merged**